public_html/wp-content/plugins/wordcamp-speaker-feedback/includes/comment.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace WordCamp\SpeakerFeedback\Comment;
+
+use WP_User;
+
+defined( 'WPINC' ) || die();
+
+const COMMENT_TYPE = 'speaker-feedback';
+
+/**
+ * Add a new feedback submission.
+ *
+ * @param int       $post_id         The ID of the post to attach the feedback to.
+ * @param array|int $feedback_author Either an array containing 'name' and 'email' values, or a user ID.
+ * @param array     $feedback_meta   An associative array of key => value pairs.
+ *
+ * @return false|int
+ */
+function add_feedback( $post_id, $feedback_author, array $feedback_meta ) {
+	$args = array(
+		'comment_approved' => 0, // "hold".
+		'comment_post_ID'  => $post_id,
+		'comment_type'     => COMMENT_TYPE,
+		'comment_meta'     => $feedback_meta,
```

Comment:
Maybe when we figure out what the meta we need is, we can add an abstraction here to sanitize/format & prefix the meta keys & values… (nothing to do now, just thinking out loud)

Reply:
Yeah, I was thinking we'd do a separate file with all the meta-related functionality, but it might make sense here to run the data through a normalizing/sanitizing function once we have that available.